corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -104,7 +104,9 @@ def process_changes(self, since, forever):
                         self._record_change_exception_in_datadog(change)
                         raise
                     else:
-                        self.fire_change_processed_event(change, context)
+                        updated = self.fire_change_processed_event(change, context)
+                        if updated:
+                            self._record_checkpoint_in_datadog()
```

Comment:
Not sure where you mean but most other places you might put this don't have all the context required.